Add support for Zengge Bluetooth bulbs

mjg59 opened the pull request:
Adds support for the Zengge Bluetooth RGBW bulbs. These are sold under a
number of brands, including Flux. The bulbs do not support full RGBW
control - they turn off the RGB LEDs when white is enabled, and vice versa.

**Description:**


**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#<home-assistant.github.io PR number goes here>

**Example entry for `configuration.yaml` (if applicable):**
```yaml
  - platform: zengge
    devices:
      C4:BE:84:51:54:8B:
        name: Living Room
```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51

amahlaka commented:
@mjg59 take a look here, maybe we could work both of these out #4043

amahlaka commented:
@balloob this seems to be nicer version of the 1 i tried to make

mjg59 commented:
It's similar, but I think they're different protocols - your Mipow bulb is setting rgb with a simple three byte command, the Zengge bulbs have a more complicated stream.

amahlaka commented:
yeah, here is the protocol https://github.com/Phhere/Playbulb/blob/master/protocols/smart.md

amahlaka commented:
hmm, now im wondering how to find the uuid for the color handle

amahlaka commented:
are you sure that your config example is valid, i dont think that ha likes those : in the address

amahlaka commented:
Please make a documentation page for  home-assistant.github.io

amahlaka commented:
@mjg59 you may want to reun the travis as well

mjg59 commented:
@amahlaka Seems to work fine for me with the address in that format

mjg59 commented:
retest this please

mjg59 commented:
(Does travis support that hook from github? Guess I'll find out…)

arsaboo commented:
It may be better to use something like below to be consistent.
```
  - platform: zengge
    devices:
      address: C4:BE:84:51:54:8B
      name: Living Room
```

mjg59 commented:
@arsaboo This seems consistent with other lighting drivers, such as https://home-assistant.io/components/light.flux_led/

robbiet480 commented:
@mjg59 Travis will not automatically retest without you pushing code but I will manually kick it off again.

arsaboo commented:
@mjg59 Good point....makes sense!

amahlaka commented:
@mjg59 well, the real problem is that if you have those ":" in there like that it wont work, arsaboo's suggestion is better,

amahlaka commented:
the reason why flux works is that its . instead of :

arsaboo commented:
@amahlaka The flux component follows the same structure as used by @mjg59...so I guess it is fine.

amahlaka commented:
no, the difference is that in FLUX it is a ip address using dots (.) but here we are using Colon (:) which is used for the value mapping in config file

Danielhiversen commented:
We are using the same format for the Broadlink component: https://home-assistant.io/components/switch.broadlink/

amahlaka commented:
still, in all cases if there is a mac addreess with colons, it is always inside single or double quotes

amahlaka commented:
- platform: zengge
    devices:
      C4:BE:84:51:54:8B:
        name: Living Room
if done like this, it sets C4 to BE same way that name is set to living room, we could replace colons with -

mjg59 commented:
@amahlaka The Tikteck component uses : in this way, and I'm running it without problems.

amahlaka commented:
@mjg59 i get this error: TypeError: __init__() missing 1 required positional argument: 'name'
```
light:
  - platform: zengge
    devices:
      55:37:4B:12:AC:E6:
        name: room
```

amahlaka commented:
@mjg59 have you updated the requirements by running the script?

Commit:
Add support for Zengge Bluetooth bulbs

Adds support for the Zengge Bluetooth RGBW bulbs. These are sold under a
number of brands, including Flux. The bulbs do not support full RGBW
control - they turn off the RGB LEDs when white is enabled, and vice versa.

balloob committed:
Update zengge.py

balloob commented:
Looks good! :dolphin:
